submissions/TaniaBondarenko/frogger_game/app.js

```diff
@@ -2,62 +2,76 @@ const borders = {
     rightBorder: 400,
     leftBorder: 5,
     upBorder: 410,
-    downBorder: 70
+    downBorder: 5
 };
+const someGap = 10;
 
-const Enemy = function (x, y) {
-    this.sprite = 'images/enemy-bug.png';
-    this.x = x;
-    this.y = y;
-    this.speed = 75;
-
-    this.render = function () {
+class GameSprite {
+    constructor (x, y,sprite)  {
+        this.sprite = sprite;
+        this.x = x;
+        this.y = y;
+    }
+    render = function () {
         ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
     };
+}
+
+class Enemy extends GameSprite {
+    constructor(x, y, sprite = "images/enemy-bug.png", speed = "75") {
+        super(x, y, sprite);
+        this.x = x;
+        this.y = y;
+        this.speed = speed;
+        this.sprite = sprite;
+    }
+    render(){};
 };
 
     Enemy.prototype.update = function (dt) {
```

Comment:
if we are using ES6 classes instead of functional prototype definition it's better to move this definition inside the 'Enemy' class code like this:
```
update(dt) {
...
} 
```
In the same way as other methods like 'constructor' or 'update', that we have inside of the 'Enemy' class definition.
Using a prototype is a functional way.

Reply:
Okay. And what about long functions? Do we handle it the same way, including into class?

Reply:
Why not?  The size of the function code doesn't matter in this case.  
There is only one important thing: this function is a method of this class.  
So this method should be included in this class definition.

Reply:
Got it! Thank You for clarification